### ISSUE TITLE
Doxygen doesn't process markdown tables correctly

### ISSUE THREAD

**Issue by dvdsosa** [doxygen/doxygen#7113]
### Issue

After documentation is built, markdown tables are not processed correctly. The way I figured this out, is because of `\snippet` not working.

### Example
In this code

```
/*!
 * \page page1 Example
 *
 * | ID    | Defined in:                              |
 * |------ | -----------------------------------------|
 * | row 1 | Untouched, zero at out of reset.         |
 * | row 2 | \snippet port.c Priority PENDSV_PRI      |
 *
 */
```
`\snippet` is not working because doxygen is adding a `<\td>` to snippet block_id as doxygen outputs states:

> warning: block marked with [Priority PENDSV_PRI <\/td>] for \snippet should appear twice in file port.c, found it 0 times

### Workaround

This is happening in doxygen -v 1.8.15 but did not in 1.8.13. The ways I have to work around this is adding `<\td>` to the [block_id] markers in the source file. Another way should be using html tables instead markdown.

I have not found this issue before in here. I appreciate if developers could have a look to the source code and why this is happening.

Kind regards.

**Comment by albert-github** [comment]
You refer to a snippet block but there is no information about e.g. this snippet block. 
- Can you please attach a, small, self contained example (source+configuration file in a tar or zip) that allows us to reproduce the problem?  Please don't add external links as they might not be persistent.

**Comment by dvdsosa** [comment]
Hi @albert-github, for sure. 

Here I attach an example. I think I'm referencing the snippet as doc says, but if you could have a look would be great. Thanks.
[doxygentest.zip](https://github.com/doxygen/doxygen/files/3377454/doxygentest.zip)

**Comment by albert-github** [comment]
I've just pushed a proposed patch, pull request #7114

**Comment by dvdsosa** [comment]
> Have to dive into the problem as when adding:
> 
> ```
>  *
>  * \snippet port.c Priority PENDSV_PRI
>  *
>  * \snippet port.c Priority SYSTICK_PRI
> ```
> 
> at the end of the `.h` file the snippets are shown there correctly but still not in the table. Looks like the entire text is not included and it looks like that the name of the snippet is not determined properly (it adds the ` </td>` to it).
> 
> Small note in a minimal file it would be nice also not to have the `.git` directory etc.

Thank you! I have updated the zip file to remove .git directory etc.

**Comment by albert-github** [comment]
Code has been integrated in master on github (please don't close the issue as this will be done at the moment of a release).

**Comment by doxygen** [comment]
This issue was previously marked 'fixed but not released',
which means it should be fixed in doxygen version 1.8.16.
Please verify if this is indeed the case. Reopen the
issue if you think it is not fixed and please include any additional information
that you think can be relevant (preferably in the form of a self-contained example).
